modules/menu/menu.c

```diff
@@ -407,6 +407,31 @@ static int cmd_ua_find(struct re_printf *pf, void *arg)
 }
 
 
+static int cmd_ua_delete(struct re_printf *pf, void *arg)
+{
+	const struct cmd_arg *carg = arg;
+	struct ua *ua = NULL;
+
+	if (str_isset(carg->prm)) {
+		ua = uag_find_aor(carg->prm);
+	}
+
+	if (!ua) {
+		return ENOENT;
+	}
+
+	if (ua == uag_cur()) {
+		(void)cmd_ua_next(pf, NULL);
+	}
+
+	(void)re_hprintf(pf, "deleting ua: %s\n", carg->prm);
+	mem_deref(ua);
+	(void)ua_print_reg_status(pf, NULL);
+
+	return 0;
+}
+
+
```

Comment:
this block looks fine. keep in mind that if the last UA is deleted, the current ua pointer
will point to an invalid object.

Reply:
True. It would be better to set it to `NULL` in that case, however when the list is empty then `uag_current()` will return `NULL` so this shouldn't be a problem. It would require more logic to implement setting it to `NULL`.

Reply:
What would you think about this:
```
diff --git a/modules/menu/menu.c b/modules/menu/menu.c
index 752ef62..dee80f8 100644
--- a/modules/menu/menu.c
+++ b/modules/menu/menu.c
@@ -367,8 +367,10 @@ static int cmd_ua_next(struct re_printf *pf, void *unused)
 
        if (!menu.le_cur)
                menu.le_cur = list_head(uag_list());
-       if (!menu.le_cur)
+       if (!menu.le_cur) {
+               uag_current_set(NULL);
                return 0;
+       }
 
        menu.le_cur = menu.le_cur->next ?
                menu.le_cur->next : list_head(uag_list());
```